python/ray/ml/result.py

```diff
@@ -4,15 +4,29 @@
 
 
 class Result:
-    """The final result of a ML training run.
+    """The final result of a ML training run or a Tune trial.
 
-    This is the class produced by Trainer.fit(). It contains a checkpoint, which
-    can be used for resuming training and for creating a Predictor object. It also
-    contains a metrics object describing training metrics.
+    This is the class produced by Trainer.fit() or Tuner.fit() (through ResultGrid).
+    It contains a checkpoint, which can be used for resuming training and for
+    creating a Predictor object. It also contains a metrics object describing
+    training metrics. `status` and `error` are included so that non successful runs
+    and trials can be represented as well.
+
+    Args:
+        metrics: The final metrics as reported by an Trainable.
+        checkpoint: The final checkpoint of the Trainable
+        error: The execution error of the Trainable run, if the trial finishes in error.
     """
 
-    checkpoint: Optional[Checkpoint]
-    metrics: Any
+    def __init__(
+        self,
+        metrics: Any,
+        checkpoint: Optional[Checkpoint],
+        error: Optional[str] = None,
```

Comment:
should this just be str? Maybe an actual exception or enum with possible statuses?

Also, is this meant to be an internal thing only? If not, exposing an `error` seems off.

Reply:
Max, can you say whether this concern is blocking for merging or can be revised at a later stage?

Reply:
@amogkam As this is mostly brought up to be consumed by Train (at least for now). Can you take a look?

Reply:
Per offline discussion, we can make this a `Optional[Exception]`. We can follow up by propagating the root cause error better in Tune.